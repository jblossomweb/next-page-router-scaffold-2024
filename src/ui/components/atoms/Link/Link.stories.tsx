import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import Link, { type LinkProps } from './Link'

export default {
  title: 'components/atoms/Link',
  component: Link,
  argTypes: {
    external: { control: 'boolean' },
    href: { control: 'text' },
    children: { control: 'text' },
  },
} as Meta<typeof Link>

const Template: StoryFn<typeof Link> = (args: LinkProps) => <Link {...args} />

export const Internal = Template.bind({})
Internal.args = {
  href: '/route',
  children: 'Link to Internal Route',
}

export const External = Template.bind({})
External.args = {
  external: true,
  href: 'https://www.google.com/',
  children: 'Link to External Url',
}

export const NoChildren = Template.bind({})
NoChildren.args = {
  href: 'https://www.google.com/',
}
