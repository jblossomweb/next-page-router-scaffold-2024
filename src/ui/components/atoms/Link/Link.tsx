import React from 'react'
import NextLink from 'next/link'

export interface LinkProps {
  href: string
  external?: boolean
  children?: React.ReactNode
}

const Link: React.FC<LinkProps> = ({
  href, //
  external = false,
  children = '',
}) =>
  external ? (
    <a
      href={href}
      target="_blank"
      rel="noreferrer"
    >
      {children}
    </a>
  ) : (
    <NextLink href={href}>{children}</NextLink>
  )

export default Link
