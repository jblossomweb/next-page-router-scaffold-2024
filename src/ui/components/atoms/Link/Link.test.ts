import storySnaps from 'test/utils/storySnaps'
import * as stories from './Link.stories'

const path = __dirname.split('/src/')[1]

describe(path, () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
