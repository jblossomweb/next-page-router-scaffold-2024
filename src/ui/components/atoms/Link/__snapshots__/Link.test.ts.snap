// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/components/atoms/Link stories External matches markup snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://www.google.com/"
    rel="noreferrer"
    target="_blank"
  >
    Link to External Url
  </a>
</DocumentFragment>
`;

exports[`ui/components/atoms/Link stories Internal matches markup snapshot 1`] = `
<DocumentFragment>
  <a
    href="/route"
  >
    Link to Internal Route
  </a>
</DocumentFragment>
`;

exports[`ui/components/atoms/Link stories NoChildren matches markup snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://www.google.com/"
  />
</DocumentFragment>
`;
